packages/dev/core/src/Cameras/freeCamera.ts

```diff
@@ -376,36 +376,47 @@ export class FreeCamera extends TargetCamera {
             actualDisplacement = displacement.add(this.getScene().gravity);
         }
 
-        coordinator.getNewPosition(this._oldPosition, actualDisplacement, this._collider, 3, null, this._onCollisionPositionChange, this.uniqueId);
+        coordinator.getNewPosition(
+            this._oldPosition,
+            actualDisplacement,
+            this._collider,
+            3,
+            null,
+            (_collisionId: number, newPosition: Vector3, collidedMesh: Nullable<AbstractMesh> = null) => {

```

Comment:
Isnt this creating more GC ?

Reply:
There is no way to pass the boolean to the function, if used as a callback. That would require changing the collider's signature as well.

Reply:
the mode could be a property of the class instead ? as it is always the same per camera type ?

Reply:
sure, it can. can be a flag on the class level instead of being passed as is it now

Reply:
Done :-)